### PR TITLE
Work in progress: Audio recording

### PR DESCRIPTION
**Pull request by jamesbvaughan**
This branch adds the ability to record audio for use in the Speech class.

NOTE: **This is not yet finished so do not merge it yet.**

### CODE REVIEW

File: src/main/java/com/auroraapi/models/Audio.java [hunk 1]
```diff
@@ -1,4 +1,80 @@
...
+  ByteArrayOutputStream data = new ByteArrayOutputStream();
+
+  void start() {
+    try {
+      DataLine.Info info = new DataLine.Info(TargetDataLine.class, format);
+
+      // checks if system supports the data line
+      if (!AudioSystem.isLineSupported(info)) {
+        System.out.println("Line not supported");
+        System.exit(0);
```
Comment (vanshg) [inline src/main/java/com/auroraapi/models/Audio.java]: I don't think we should exit (it'll be unexpected to whatever developer is using the SDK). We should throw an exception instead or return a boolean indicating success or something so the developer can handle the case when there's no audio hardware present

File: src/main/java/com/auroraapi/models/Audio.java [hunk 2]
```diff
@@ -1,4 +1,80 @@
...
+  /**
+   * Closes the target data line to finish capturing and recording
+   */
+  void finish() {
+    line.stop();
+    line.close();
+    System.out.println("Finished");
+  }
+
+  public static Audio record(int length, float silence_length) {
```
Comment (vanshg) [inline src/main/java/com/auroraapi/models/Audio.java]: camel case is the norm in Java for variable names

File: src/main/java/com/auroraapi/models/Speech.java [hunk 3]
```diff
@@ -14,4 +14,8 @@ public Audio getAudio() {
     public void setAudio(Audio audio) {
         this.audio = audio;
     }
+
+    public static Speech listen(int length, float silence_len) {
```
Comment (vanshg) [inline src/main/java/com/auroraapi/models/Speech.java]: camel case here as well

File: src/main/java/com/auroraapi/models/Audio.java [hunk 4]
```diff
@@ -1,4 +1,80 @@
 package com.auroraapi.models;
 
-public class Audio {
-}
+import javax.sound.sampled.*;
+import java.io.*;
+
+class Audio {
+  static final int NUM_CHANNELS = 1;
```
Comment (vanshg) [inline src/main/java/com/auroraapi/models/Audio.java]: Let's also make all the class variables `private`

**Review by vanshg — CHANGES_REQUESTED** [review]

File: src/main/java/com/auroraapi/models/Audio.java (hunk 1 above)
Comment (jamesbvaughan) [reply src/main/java/com/auroraapi/models/Audio.java]: Good call. Half of this was copy/pasted from an example and I didn't really think about all of it.

**Review by vanshg — APPROVED** [review]

**Comment by vanshg** [comment]
@jamesbvaughan I think this is in a good state now. We can create a separate PR for silence related recording and continuous recording. Go ahead and merge if you think it's good

**Pull request merged**